### DISCUSSION TITLE
How do I use HardNegativeSampler ?

### DISCUSSION THREAD

**Discussion by DSRajesh** [Project-MONAI/MONAI#7135] [Q&A]
Hello
How do I use HardNegativeSampler   while training a classifier using DenseNet121 ?

Thanks
Rajesh

**Comment by KumoLiu** [comment]
Hi @DSRajesh, here is the training workflow.
https://github.com/Project-MONAI/MONAI/blob/0a3cdd9f1403cbad057b96230840ee17cf430dbf/monai/apps/detection/utils/hard_negative_sampler.py#L104-L108
So you can put `HardNegativeSampler` after `out = model(input)`.

You can also refer to:
https://github.com/Project-MONAI/MONAI/blob/0a3cdd9f1403cbad057b96230840ee17cf430dbf/monai/apps/detection/networks/retinanet_detector.py#L938-L956
Thanks!
